### PR TITLE
Upgrade React dependencies to fix compile issues

### PR DESCRIPTION
**Pull request by mraible**
After this change, there's still two warnings when generating a new project:

```
WARNING in ./src/main/webapp/app/modules/administration/metrics/metrics.tsx
[127, 97]: Lambdas are forbidden in JSX attributes due to their rendering performance impact

 @ ./src/main/webapp/app/modules/administration/index.tsx 7:16-44
 @ ./src/main/webapp/app/routes.tsx
 @ ./src/main/webapp/app/app.tsx
 @ ./src/main/webapp/app/index.tsx
 @ multi react-hot-loader/patch ./src/main/webapp/app/index

WARNING in ./src/main/webapp/app/modules/administration/health/health.tsx
[89, 37]: Lambdas are forbidden in JSX attributes due to their rendering performance impact

 @ ./src/main/webapp/app/modules/administration/index.tsx 6:15-41
 @ ./src/main/webapp/app/routes.tsx
 @ ./src/main/webapp/app/app.tsx
 @ ./src/main/webapp/app/index.tsx
 @ multi react-hot-loader/patch ./src/main/webapp/app/index
```

My `.yo-rc.json` file:

```json
{
  "generator-jhipster": {
    "promptValues": {
      "packageName": "com.mycompany.myapp",
      "nativeLanguage": "en"
    },
    "jhipsterVersion": "4.10.2",
    "baseName": "devoxx2",
    "packageName": "com.mycompany.myapp",
    "packageFolder": "com/mycompany/myapp",
    "serverPort": "8080",
    "authenticationType": "jwt",
    "hibernateCache": "ehcache",
    "clusteredHttpSession": false,
    "websocket": false,
    "databaseType": "sql",
    "devDatabaseType": "h2Disk",
    "prodDatabaseType": "mysql",
    "searchEngine": false,
    "messageBroker": false,
    "serviceDiscoveryType": false,
    "buildTool": "maven",
    "enableSocialSignIn": false,
    "enableSwaggerCodegen": false,
    "jwtSecretKey": "ba3e5fbb101b99d8ab5c6f68def162db0efb7969",
    "clientFramework": "react",
    "useSass": false,
    "clientPackageManager": "yarn",
    "applicationType": "monolith",
    "testFrameworks": [],
    "jhiPrefix": "jhi",
    "enableTranslation": true,
    "nativeLanguage": "en",
    "languages": [
      "en",
      "nl"
    ]
  }
}
```

### CODE REVIEW

**Comment by mraible** [comment]
Strange - it looks like my merge from master caused this PR to pull in a punch of comments on other commits. I'll rebase and see if I can fix. 

FWIW, after generating an app and trying to run it, I'm unable to login. There's just a blank screen and the following in my console:

```
Modal.js?2a72:220 Uncaught TypeError: Cannot read property 'parentNode' of undefined
    at Modal.renderIntoSubtree (Modal.js?2a72:220)
    at Modal.show (Modal.js?2a72:184)
    at Modal.togglePortal (Modal.js?2a72:138)
    at Modal.componentDidMount (Modal.js?2a72:78)
    at commitLifeCycles (react-dom.development.js?d6fa:11505)
    at commitAllLifeCycles (react-dom.development.js?d6fa:12294)
    at HTMLUnknownElement.callCallback (react-dom.development.js?d6fa:1299)
    at Object.invokeGuardedCallbackDev (react-dom.development.js?d6fa:1338)
    at invokeGuardedCallback (react-dom.development.js?d6fa:1195)
    at commitAllWork (react-dom.development.js?d6fa:12415)
renderIntoSubtree @ Modal.js?2a72:220
show @ Modal.js?2a72:184
togglePortal @ Modal.js?2a72:138
componentDidMount @ Modal.js?2a72:78
commitLifeCycles @ react-dom.development.js?d6fa:11505
commitAllLifeCycles @ react-dom.development.js?d6fa:12294
callCallback @ react-dom.development.js?d6fa:1299
invokeGuardedCallbackDev @ react-dom.development.js?d6fa:1338
invokeGuardedCallback @ react-dom.development.js?d6fa:1195
commitAllWork @ react-dom.development.js?d6fa:12415
workLoop @ react-dom.development.js?d6fa:12687
callCallback @ react-dom.development.js?d6fa:1299
invokeGuardedCallbackDev @ react-dom.development.js?d6fa:1338
invokeGuardedCallback @ react-dom.development.js?d6fa:1195
performWork @ react-dom.development.js?d6fa:12800
batchedUpdates @ react-dom.development.js?d6fa:13244
performFiberBatchedUpdates @ react-dom.development.js?d6fa:1646
stackBatchedUpdates @ react-dom.development.js?d6fa:1637
batchedUpdates @ react-dom.development.js?d6fa:1651
batchedUpdatesWithControlledComponents @ react-dom.development.js?d6fa:1664
dispatchEvent @ react-dom.development.js?d6fa:1874
react-dom.development.js?d6fa:8305 The above error occurred in the <Modal> component:
    in Modal (created by LoginModal)
    in LoginModal (created by Login)
    in Login (created by Connect(Login))
    in Connect(Login) (created by Route)
    in Route (created by Routes)
    in div (created by Routes)
    in Routes (created by App)
    in div (created by Card)
    in Card (created by App)
    in div (created by App)
    in div (created by App)
    in Router (created by HashRouter)
    in HashRouter (created by App)
    in App (created by Connect(App))
    in Connect(App)
    in div
    in Provider
    in AppContainer

React will try to recreate this component tree from scratch using the error boundary you provided, AppContainer.
logCapturedError$1 @ react-dom.development.js?d6fa:8305
captureError @ react-dom.development.js?d6fa:12993
commitAllWork @ react-dom.development.js?d6fa:12423
workLoop @ react-dom.development.js?d6fa:12687
callCallback @ react-dom.development.js?d6fa:1299
invokeGuardedCallbackDev @ react-dom.development.js?d6fa:1338
invokeGuardedCallback @ react-dom.development.js?d6fa:1195
performWork @ react-dom.development.js?d6fa:12800
batchedUpdates @ react-dom.development.js?d6fa:13244
performFiberBatchedUpdates @ react-dom.development.js?d6fa:1646
stackBatchedUpdates @ react-dom.development.js?d6fa:1637
batchedUpdates @ react-dom.development.js?d6fa:1651
batchedUpdatesWithControlledComponents @ react-dom.development.js?d6fa:1664
dispatchEvent @ react-dom.development.js?d6fa:1874
AppContainer.dev.js?fe74:83 Error: A cross-origin error was thrown. React doesn't have access to the actual error object in development. See https://fb.me/react-crossorigin-error for more information.
    at Object.invokeGuardedCallbackDev (react-dom.development.js?d6fa:1345)
    at invokeGuardedCallback (react-dom.development.js?d6fa:1195)
    at commitAllWork (react-dom.development.js?d6fa:12415)
    at workLoop (react-dom.development.js?d6fa:12687)
    at HTMLUnknownElement.callCallback (react-dom.development.js?d6fa:1299)
    at Object.invokeGuardedCallbackDev (react-dom.development.js?d6fa:1338)
    at invokeGuardedCallback (react-dom.development.js?d6fa:1195)
    at performWork (react-dom.development.js?d6fa:12800)
    at batchedUpdates (react-dom.development.js?d6fa:13244)
    at performFiberBatchedUpdates (react-dom.development.js?d6fa:1646)
render @ AppContainer.dev.js?fe74:83
finishClassComponent @ react-dom.development.js?d6fa:10249
updateClassComponent @ react-dom.development.js?d6fa:10226
beginWork @ react-dom.development.js?d6fa:10605
performUnitOfWork @ react-dom.development.js?d6fa:12573
workLoop @ react-dom.development.js?d6fa:12682
callCallback @ react-dom.development.js?d6fa:1299
invokeGuardedCallbackDev @ react-dom.development.js?d6fa:1338
invokeGuardedCallback @ react-dom.development.js?d6fa:1195
performWork @ react-dom.development.js?d6fa:12800
batchedUpdates @ react-dom.development.js?d6fa:13244
performFiberBatchedUpdates @ react-dom.development.js?d6fa:1646
stackBatchedUpdates @ react-dom.development.js?d6fa:1637
batchedUpdates @ react-dom.development.js?d6fa:1651
batchedUpdatesWithControlledComponents @ react-dom.development.js?d6fa:1664
dispatchEvent @ react-dom.development.js?d6fa:1874
Modal.js?2a72:220 Uncaught TypeError: Cannot read property 'parentNode' of undefined
    at Modal.renderIntoSubtree (Modal.js?2a72:220)
    at Modal.show (Modal.js?2a72:184)
    at Modal.togglePortal (Modal.js?2a72:138)
    at Modal.componentDidMount (Modal.js?2a72:78)
    at commitLifeCycles (react-dom.development.js?d6fa:11505)
    at commitAllLifeCycles (react-dom.development.js?d6fa:12294)
    at HTMLUnknownElement.callCallback (react-dom.development.js?d6fa:1299)
    at Object.invokeGuardedCallbackDev (react-dom.development.js?d6fa:1338)
    at invokeGuardedCallback (react-dom.development.js?d6fa:1195)
    at commitAllWork (react-dom.development.js?d6fa:12415)
renderIntoSubtree @ Modal.js?2a72:220
show @ Modal.js?2a72:184
togglePortal @ Modal.js?2a72:138
componentDidMount @ Modal.js?2a72:78
commitLifeCycles @ react-dom.development.js?d6fa:11505
commitAllLifeCycles @ react-dom.development.js?d6fa:12294
callCallback @ react-dom.development.js?d6fa:1299
invokeGuardedCallbackDev @ react-dom.development.js?d6fa:1338
invokeGuardedCallback @ react-dom.development.js?d6fa:1195
commitAllWork @ react-dom.development.js?d6fa:12415
workLoop @ react-dom.development.js?d6fa:12687
callCallback @ react-dom.development.js?d6fa:1299
invokeGuardedCallbackDev @ react-dom.development.js?d6fa:1338
invokeGuardedCallback @ react-dom.development.js?d6fa:1195
performWork @ react-dom.development.js?d6fa:12800
batchedUpdates @ react-dom.development.js?d6fa:13244
performFiberBatchedUpdates @ react-dom.development.js?d6fa:1646
stackBatchedUpdates @ react-dom.development.js?d6fa:1637
batchedUpdates @ react-dom.development.js?d6fa:1651
batchedUpdatesWithControlledComponents @ react-dom.development.js?d6fa:1664
dispatchEvent @ react-dom.development.js?d6fa:1874
react-dom.development.js?d6fa:8305 The above error occurred in the <Modal> component:
    in Modal (created by LoginModal)
    in LoginModal (created by Login)
    in Login (created by Connect(Login))
    in Connect(Login) (created by Route)
    in Route (created by Routes)
    in div (created by Routes)
    in Routes (created by App)
    in div (created by Card)
    in Card (created by App)
    in div (created by App)
    in div (created by App)
    in Router (created by HashRouter)
    in HashRouter (created by App)
    in App (created by Connect(App))
    in Connect(App)
    in div
    in Provider
    in AppContainer

This error was initially handled by the error boundary AppContainer.
Recreating the tree from scratch failed so React will unmount the tree.
logCapturedError$1 @ react-dom.development.js?d6fa:8305
captureError @ react-dom.development.js?d6fa:12993
commitAllWork @ react-dom.development.js?d6fa:12423
workLoop @ react-dom.development.js?d6fa:12687
callCallback @ react-dom.development.js?d6fa:1299
invokeGuardedCallbackDev @ react-dom.development.js?d6fa:1338
invokeGuardedCallback @ react-dom.development.js?d6fa:1195
performWork @ react-dom.development.js?d6fa:12800
batchedUpdates @ react-dom.development.js?d6fa:13244
performFiberBatchedUpdates @ react-dom.development.js?d6fa:1646
stackBatchedUpdates @ react-dom.development.js?d6fa:1637
batchedUpdates @ react-dom.development.js?d6fa:1651
batchedUpdatesWithControlledComponents @ react-dom.development.js?d6fa:1664
dispatchEvent @ react-dom.development.js?d6fa:1874
logger-middleware.ts?6a25:6 loading-bar/SHOW
logger-middleware.ts?6a25:6 authentication/GET_SESSION_PENDING
react-dom.development.js?d6fa:1345 Uncaught Error: A cross-origin error was thrown. React doesn't have access to the actual error object in development. See https://fb.me/react-crossorigin-error for more information.
    at Object.invokeGuardedCallbackDev (react-dom.development.js?d6fa:1345)
    at invokeGuardedCallback (react-dom.development.js?d6fa:1195)
    at commitAllWork (react-dom.development.js?d6fa:12415)
    at workLoop (react-dom.development.js?d6fa:12687)
    at HTMLUnknownElement.callCallback (react-dom.development.js?d6fa:1299)
    at Object.invokeGuardedCallbackDev (react-dom.development.js?d6fa:1338)
    at invokeGuardedCallback (react-dom.development.js?d6fa:1195)
    at performWork (react-dom.development.js?d6fa:12800)
    at batchedUpdates (react-dom.development.js?d6fa:13244)
    at performFiberBatchedUpdates (react-dom.development.js?d6fa:1646)
invokeGuardedCallbackDev @ react-dom.development.js?d6fa:1345
invokeGuardedCallback @ react-dom.development.js?d6fa:1195
commitAllWork @ react-dom.development.js?d6fa:12415
workLoop @ react-dom.development.js?d6fa:12687
callCallback @ react-dom.development.js?d6fa:1299
invokeGuardedCallbackDev @ react-dom.development.js?d6fa:1338
invokeGuardedCallback @ react-dom.development.js?d6fa:1195
performWork @ react-dom.development.js?d6fa:12800
batchedUpdates @ react-dom.development.js?d6fa:13244
performFiberBatchedUpdates @ react-dom.development.js?d6fa:1646
stackBatchedUpdates @ react-dom.development.js?d6fa:1637
batchedUpdates @ react-dom.development.js?d6fa:1651
batchedUpdatesWithControlledComponents @ react-dom.development.js?d6fa:1664
dispatchEvent @ react-dom.development.js?d6fa:1874
xhr.js?ae5f:179 GET http://localhost:8000/api/account 401 (Unauthorized)
```

**Comment by mraible** [comment]
@deepu105 I tried to rebase to eliminate commits/comments on this PR, but failed. I don't know enough about the React commits (local vs server) to fix. Maybe I should just always accept local changes?

**Comment by mraible** [comment]
@deepu105 If I generate a project w/o i18n, it doesn't compile.

```
ERROR in ./src/main/webapp/app/modules/account/settings/settings.tsx
Module not found: Error: Can't resolve '../../../config/translation' in '/Users/mraible/devoxx2/src/main/webapp/app/modules/account/settings'
 @ ./src/main/webapp/app/modules/account/settings/settings.tsx 25:20-58
 @ ./src/main/webapp/app/modules/account/index.tsx
 @ ./src/main/webapp/app/routes.tsx
 @ ./src/main/webapp/app/app.tsx
 @ ./src/main/webapp/app/index.tsx
 @ multi react-hot-loader/patch ./src/main/webapp/app/index

ERROR in ./src/main/webapp/app/index.tsx
Module not found: Error: Can't resolve './config/translation' in '/Users/mraible/devoxx2/src/main/webapp/app'
 @ ./src/main/webapp/app/index.tsx 10:20-51
 @ multi react-hot-loader/patch ./src/main/webapp/app/index

ERROR in ./src/main/webapp/app/reducers/index.ts
Module not found: Error: Can't resolve './locale' in '/Users/mraible/devoxx2/src/main/webapp/app/reducers'
 @ ./src/main/webapp/app/reducers/index.ts 5:15-34
 @ ./src/main/webapp/app/config/store.ts
 @ ./src/main/webapp/app/index.tsx
 @ multi react-hot-loader/patch ./src/main/webapp/app/index

ERROR in ./src/main/webapp/app/app.tsx
Module not found: Error: Can't resolve './reducers/locale' in '/Users/mraible/devoxx2/src/main/webapp/app'
 @ ./src/main/webapp/app/app.tsx 19:15-43
 @ ./src/main/webapp/app/index.tsx
 @ multi react-hot-loader/patch ./src/main/webapp/app/index

ERROR in [at-loader] ./src/main/webapp/app/app.tsx:9:27
    TS2307: Cannot find module './reducers/locale'.

ERROR in [at-loader] ./src/main/webapp/app/index.tsx:9:33
    TS2307: Cannot find module './config/translation'.

ERROR in [at-loader] ./src/main/webapp/app/modules/account/settings/settings.tsx:5:25
    TS2307: Cannot find module '../../../config/translation'.

ERROR in [at-loader] ./src/main/webapp/app/reducers/index.ts:4:20
    TS2307: Cannot find module './locale'.

ERROR in [at-loader] ./src/main/webapp/app/shared/layout/header/header.tsx:12:25
    TS2307: Cannot find module '../../../config/translation'.
error Command failed with exit code 2.
error Command failed with exit code 1.
error Command failed with exit code 1.
```

Here's the `.yo-rc.json` for a project w/o i18n.

```json
{
  "generator-jhipster": {
    "promptValues": {
      "packageName": "com.mycompany.myapp"
    },
    "jhipsterVersion": "4.10.2",
    "baseName": "devoxx2",
    "packageName": "com.mycompany.myapp",
    "packageFolder": "com/mycompany/myapp",
    "serverPort": "8080",
    "authenticationType": "jwt",
    "hibernateCache": "ehcache",
    "clusteredHttpSession": false,
    "websocket": false,
    "databaseType": "sql",
    "devDatabaseType": "h2Disk",
    "prodDatabaseType": "mysql",
    "searchEngine": false,
    "messageBroker": false,
    "serviceDiscoveryType": false,
    "buildTool": "maven",
    "enableSocialSignIn": false,
    "enableSwaggerCodegen": false,
    "jwtSecretKey": "686df9794ddfa0d57986deb6420e38446ad34dff",
    "clientFramework": "react",
    "useSass": false,
    "clientPackageManager": "yarn",
    "applicationType": "monolith",
    "testFrameworks": [],
    "jhiPrefix": "jhi",
    "enableTranslation": false
  }
}
```

**Comment by mraible** [comment]
@deepu105 [Here's a PR](https://github.com/mraible/generator-jhipster/pull/3) against this branch that will fix i18n compilation issues.

**Comment by mraible** [comment]
@deepu105 Here's [another PR](https://github.com/mraible/generator-jhipster/pull/4) that fixes test compilation and execution issues.

**Comment by deepu105** [comment]
I fixed those warning in the branch already so should be fine

**Comment by mraible** [comment]
@deepu105 Can you please merge master into `jh-react`, or rebase? Once that's done, there should only be a few changes in this PR.

File: generators/server/templates/src/main/java/package/service/_UserService.java [hunk 1]
```diff
@@ -39,7 +39,9 @@
 import <%=packageName%>.service.util.RandomUtil;
 <%_ } _%>
 import <%=packageName%>.service.dto.UserDTO;
+<%_ if (authenticationType !== 'oauth2') { _%>
```
Comment (deepu105) [inline generators/server/templates/src/main/java/package/service/_UserService.java]: @mraible are the changes in this file required? the react branch is up to date with changes in master branch so I dont understand why this change shows up here?

File: generators/client/prompts.js [hunk 2]
```diff
@@ -40,6 +40,10 @@ function askForClient(meta) {
         {
             value: 'angularX',
             name: 'Angular 4'
+        },
```
Comment (deepu105) [inline generators/client/prompts.js]: since OAuth probably doesn't work with React its better to have this only in the below optional block

File: generators/client/templates/react/_package.json [hunk 3]
```diff
@@ -8,55 +8,58 @@
...
-    "react-router-dom": "4.1.2",
-    "react-transition-group": "1.2.0",
-    "reactstrap": "5.0.0-alpha.1",
-    "redux": "3.7.1",
+    "react-jhipster": "0.1.1",
+    "react-redux": "5.0.6",
+    "react-redux-loading-bar": "2.9.3",
+    "react-router-dom": "4.2.2",
+    "react-transition-group": "2.2.1",
+    "react-translate-component": "0.14.0",
```
Comment (deepu105) [inline generators/client/templates/react/_package.json]: I think this lib is not required as translation is done in the react-jhipster lib

**Comment by deepu105** [comment]
I fixed the extra stuff here I guess. Take a look

File: generators/client/prompts.js (hunk 2 above)
Comment (mraible) [reply generators/client/prompts.js]: OK, I can move, and then move it back once I have OAuth support working. Does JWT and Session work or just JWT?

File: generators/server/templates/src/main/java/package/service/_UserService.java (hunk 1 above)
Comment (mraible) [reply generators/server/templates/src/main/java/package/service/_UserService.java]: Me neither, I'll revert changes.

**Comment by deepu105** [comment]
@mraible dont worry about this. I'll fix and merge this tomorrow. I'll make sure the branch is working with default options tomorrow when I'll also add initial entity support. I already have a working entity reducer and stuff I just need to add the front end

**Comment by mraible** [comment]
OK, I'll ignore for now and concentrate on getting my Ionic module working.

File: generators/client/prompts.js (hunk 2 above)
Comment (deepu105) [reply generators/client/prompts.js]: for now just JWT but dont worry i'lltake  care of this

**Comment by deepu105** [comment]
squash merged

**Comment by deepu105** [comment]
@mraible the `jh-react` branch now works with default options. I squash merged your PRs but there is something wrong with these branches as It doesnt show as merged somehow. But the changes are on the branch regardless.
here is a working config
```
{
  "generator-jhipster": {
    "promptValues": {
      "packageName": "com.mycompany.myapp",
      "nativeLanguage": "en"
    },
    "jhipsterVersion": "4.10.2",
    "baseName": "jhipster",
    "packageName": "com.mycompany.myapp",
    "packageFolder": "com/mycompany/myapp",
    "serverPort": "8080",
    "authenticationType": "jwt",
    "hibernateCache": "ehcache",
    "clusteredHttpSession": false,
    "websocket": false,
    "databaseType": "sql",
    "devDatabaseType": "h2Disk",
    "prodDatabaseType": "mysql",
    "searchEngine": false,
    "messageBroker": false,
    "serviceDiscoveryType": false,
    "buildTool": "maven",
    "enableSocialSignIn": false,
    "enableSwaggerCodegen": false,
    "jwtSecretKey": "1ee93086c7c014db110566bfd68e760c88f33065",
    "clientFramework": "react",
    "useSass": false,
    "clientPackageManager": "yarn",
    "applicationType": "monolith",
    "testFrameworks": [],
    "jhiPrefix": "jhi",
    "enableTranslation": true,
    "nativeLanguage": "en",
    "languages": [
      "en",
      "ar-ly"
    ]
  }
}
```
